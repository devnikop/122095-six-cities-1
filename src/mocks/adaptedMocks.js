const adaptedLoginDataMock = {
  avatarUrl: `/static/avatar/5.jpg`,
  email: `doppervily@yandex.ru`,
  id: 1,
  isPro: false,
  name: `doppervily`,
};

const adaptedOfferMock = {
  bedrooms: 1,
  city: {
    coords: [48.85661, 2.351499],
    zoom: 13,
    name: `Paris`,
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  goods: [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
  host: {
    avatar: `img/avatar-angelina.jpg`,
    id: 25,
    name: `Angelina`,
    isPro: true,
  },
  id: 1,
  images: [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
  isFavorite: true,
  isPremium: false,
  maxAdults: 3,
  place: {
    coords: [48.834610000000005, 2.364499],
    zoom: 16,
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
  price: 231,
  rating: 5,
  title: `Canal View Prinsengracht`,
  type: `room`,
};

const adaptedOffersArrayMock = [
  {
    bedrooms: 1,
    city: {
      coords: [48.85661, 2.351499],
      zoom: 13,
      name: `Paris`,
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isPro: true,
    },
    id: 1,
    images: [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
    isFavorite: true,
    isPremium: false,
    maxAdults: 3,
    place: {
      coords: [48.834610000000005, 2.364499],
      zoom: 16,
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    price: 231,
    rating: 5,
    title: `Canal View Prinsengracht`,
    type: `room`,
  },
];

const adaptedReviewsArrayMock = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`,
    id: 1,
    rating: 4,
    user: {
      avatarUrl: `img/1.png`,
      id: 4,
      isPro: false,
      name: `Max`,
    },
  },
];

export {
  adaptedLoginDataMock,
  adaptedOfferMock,
  adaptedOffersArrayMock,
  adaptedReviewsArrayMock,
};
