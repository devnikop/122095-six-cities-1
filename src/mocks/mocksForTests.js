const leafletMock = {
  icon() {
    return {
      iconSize: [30, 30],
      iconUrl: `img/pin.svg`,
    };
  },
  map() {
    return {
      setView: jest.fn(),
    };
  },
  tileLayer() {
    return {
      addTo: jest.fn(),
    };
  },
  layerGroup() {
    return {
      addTo: jest.fn(),
    };
  },
  marker() {
    return {
      addTo: jest.fn(),
    };
  },
};

const offersArrayMock = [
  {
    city: `Amsterdam`,
    coords: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    placeName: `Beautiful & luxurious apartment at great location`,
    placeType: `Apartment`,
    price: 120,
    src: `img/apartment-01.jpg`,
  },
  {
    city: `Cologne`,
    coords: [52.369553943508, 4.85309666406198],
    isPremium: false,
    placeName: `Wood and stone place`,
    placeType: `Private room`,
    price: 80,
    src: `img/room.jpg`,
  },
  {
    city: `Brussels`,
    coords: [52.3909553943508, 4.929309666406198],
    isPremium: false,
    placeName: `Canal View Prinsengracht`,
    placeType: `Apartment`,
    price: 132,
    src: `img/apartment-02.jpg`,
  },
  {
    city: `Amsterdam`,
    coords: [52.3809553943508, 4.939309666406198],
    isPremium: true,
    placeName: `Nice, cozy, warm big bed apartment`,
    placeType: `Apartment`,
    price: 180,
    src: `img/apartment-03.jpg`,
  }
];

const citiesMock = [
  {
    coords: [52.370216, 4.895168],
    name: `Amsterdam`,
  },
  {
    coords: [50.850346, 4.351721],
    name: `Brussels`,
  },
  {
    coords: [51.227741, 6.773456],
    name: `Cologne`,
  },
  {
    coords: [50.937531, 6.960279],
    name: `Dusseldorf`,
  },
  {
    coords: [53.551086, 9.993682],
    name: `Hamburg`,
  },
  {
    coords: [48.856613, 2.352222],
    name: `Paris`,
  },
];

const offerMock = {
  isPremium: true,
  placeName: `Beautiful & luxurious apartment at great location`,
  placeType: `Apartment`,
  price: 120,
  src: `img/apartment-01.jpg`,
};

export {
  citiesMock,
  leafletMock,
  offerMock,
  offersArrayMock,
};
