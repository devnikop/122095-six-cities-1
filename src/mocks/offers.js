// const offers = [
//   {
//     city: `Amsterdam`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Cologne`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Dusseldorf`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Hamburg`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Amsterdam`,
//     placeName: `Beautiful & luxurious apartment at great location`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-01.jpg`,
//     price: 120,
//     coords: [52.3909553943508, 4.85309666406198],
//   },
//   {
//     city: `Cologne`,
//     placeName: `Wood and stone place`,
//     placeType: `Private room`,
//     isPremium: false,
//     src: `img/room.jpg`,
//     price: 80,
//     coords: [52.369553943508, 4.85309666406198],
//   },
//   {
//     city: `Brussels`,
//     placeName: `Canal View Prinsengracht`,
//     placeType: `Apartment`,
//     isPremium: false,
//     src: `img/apartment-02.jpg`,
//     price: 132,
//     coords: [52.3909553943508, 4.929309666406198],
//   },
//   {
//     city: `Amsterdam`,
//     placeName: `Nice, cozy, warm big bed apartment`,
//     placeType: `Apartment`,
//     isPremium: true,
//     src: `img/apartment-03.jpg`,
//     price: 180,
//     coords: [52.3809553943508, 4.939309666406198],
//   }
// ];

// const cities = [
//   {
//     name: `Amsterdam`,
//     coords: [52.370216, 4.895168],
//   },
//   {
//     name: `Brussels`,
//     coords: [50.850346, 4.351721],
//   },
//   {
//     name: `Cologne`,
//     coords: [51.227741, 6.773456],
//   },
//   {
//     name: `Dusseldorf`,
//     coords: [50.937531, 6.960279],
//   },
//   {
//     name: `Hamburg`,
//     coords: [53.551086, 9.993682],
//   },
//   {
//     name: `Paris`,
//     coords: [48.856613, 2.352222],
//   },
//   {
//     name: `Moscow`,
//     coords: [48.856613, 2.352222],
//   },
// ];

// export {
//   offers,
//   cities
// };
