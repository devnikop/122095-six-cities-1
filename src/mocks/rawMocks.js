const rawLoginDataMock = {
  "avatar_url": `/static/avatar/5.jpg`,
  "email": `doppervily@yandex.ru`,
  "id": 1,
  "is_pro": false,
  "name": `doppervily`,
};

const rawOfferMock = {
  "bedrooms": 1,
  "city": {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
  "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
  "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
  "id": 1,
  "images": [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
  "is_favorite": true,
  "is_premium": false,
  "location": {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
  "max_adults": 3,
  "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
  "price": 231,
  "rating": 4.8,
  "title": `Canal View Prinsengracht`,
  "type": `room`
};

const rawOffersArrayMock = [
  {
    "bedrooms": 1,
    "city": {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "goods": [`Towels`, `Laptop friendly workspace`, `Baby seat`, `Washer`, `Breakfast`, `Air conditioning`],
    "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
    "id": 1,
    "images": [`https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`],
    "is_favorite": true,
    "is_premium": false,
    "location": {latitude: 48.834610000000005, longitude: 2.364499, zoom: 16},
    "max_adults": 3,
    "preview_image": `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    "price": 231,
    "rating": 4.8,
    "title": `Canal View Prinsengracht`,
    "type": `room`
  }
];

const rawReviewsArrayMock = [
  {
    "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "date": `2019-05-08T14:13:56.569Z`,
    "id": 1,
    "rating": 4,
    "user": {
      "avatar_url": `img/1.png`,
      "id": 4,
      "is_pro": false,
      "name": `Max`,
    },
  },
];

export {
  rawLoginDataMock,
  rawOfferMock,
  rawOffersArrayMock,
  rawReviewsArrayMock,
};
