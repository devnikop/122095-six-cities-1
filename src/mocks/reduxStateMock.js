import {
  citiesMock,
  offersArrayMock,
  reviewsArrayMock,
  loginDataMock,
} from './mocksForTests';

import NameSpace from '../reducer/name-spaces';

const StoreMock = {
  data: {
    activeOfferId: 0,
    currentCity: citiesMock[0],
    offers: offersArrayMock,
    sortedOffers: offersArrayMock,
  },
  favorite: {
    favoriteOffers: offersArrayMock,
  },
  review: {
    reviews: reviewsArrayMock,
  },
  user: {
    isAuthorizationRequired: false,
    loginData: loginDataMock,
  },
};

const reduxStateMock = {
  [NameSpace.DATA]: StoreMock.data,
  [NameSpace.FAVORITE]: StoreMock.favorite,
  [NameSpace.REVIEW]: StoreMock.review,
  [NameSpace.USER]: StoreMock.user,
};

export default reduxStateMock;
