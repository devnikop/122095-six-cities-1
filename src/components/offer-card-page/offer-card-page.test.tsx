import { Provider } from 'react-redux';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
// import thunk from 'redux-thunk';

import { offerMock, offersArrayMock, loginDataMock } from '../../mocks/mocksForTests';
import reduxStateMock from '../../mocks/reduxStateMock';

import OfferCardPage from './offer-card-page';

const mock = {
  match: {
    params: {
      id: 1
    }
  }
};

it.skip(`OfferCardPage renders correctly`, () => {
  const {
    match
  } = mock;

  const initialState = reduxStateMock;
  const mockStore = configureStore();
  const store = mockStore(initialState);

  const tree = renderer
    .create(<Provider store={store}>
      <OfferCardPage
        activeOfferId={0}
        handleBookmarkClick={jest.fn()}
        match={match}
        nearestOffers={offersArrayMock}
        offer={offerMock}
        user={loginDataMock}
      />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
