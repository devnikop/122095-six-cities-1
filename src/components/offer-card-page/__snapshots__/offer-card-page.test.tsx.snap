// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCardPage renders correctly 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      />
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Beautiful & luxurious apartment at great location
          </h1>
          <button
            className="property__bookmark-button  button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "72%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3.6
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            room
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            1
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            3
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            324
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          />
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            />
            <span
              className="property__user-status"
            >
              
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              3
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/9.png"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Mdgnx
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "20%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  Asdaf f234frefdsafdsfggdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-05-08T14:13:56.569Z"
                >
                  May 2019
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/4.png"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Maenx
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  Aijw3ifniknfii2222 a river by the unique lightness of Amsterdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-05-08T14:13:56.569Z"
                >
                  May 2019
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="img/1.png"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Max
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "80%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-05-08T14:13:56.569Z"
                >
                  May 2019
                </time>
              </div>
            </li>
          </ul>
          <form
            action="#"
            className="reviews__form form"
            method="post"
            onChange={[Function]}
            onSubmit={[Function]}
          >
            <fieldset
              className="form__fieldset"
              disabled={false}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden"
                  id="5-stars"
                  name="rating"
                  type="radio"
                  value="5"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="4-stars"
                  name="rating"
                  type="radio"
                  value="4"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="3-stars"
                  name="rating"
                  type="radio"
                  value="3"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="2-stars"
                  name="rating"
                  type="radio"
                  value="2"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden"
                  id="1-star"
                  name="rating"
                  type="radio"
                  value="1"
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-star"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                id="review"
                name="review"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
                value=""
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </fieldset>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <div
        id="map"
        style={
          Object {
            "height": "100%",
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="cities__card place-card"
        >
          <div
            className="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://es31-server.appspot.com/six-cities/static/hotel/19.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className=" place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  324
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button  button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "72%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                onClick={[Function]}
              >
                Beautiful & luxurious apartment at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              room
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
